business/namespaces.go

```diff
@@ -404,7 +404,11 @@ func (in *NamespaceService) GetNamespacesForCluster(ctx context.Context, cluster
 
 	clusterNamespaces := []models.Namespace{}
 	for _, ns := range tokenNamespaces {
-		if ns.Cluster == cluster {
+		// TODO: The RHS of this expression shouldn't be necessary. I don't think we shold have a DefaultClusterID
+		// or a HomeClusterName constant.  We should always be using the configured cluster name (which should be
+		// set the proper "Kubernetes" default as needed).  The RHS is here because there are times when the client
+		// is using one value and the server is using the other.
+		if ns.Cluster == cluster || (cluster == DefaultClusterID && ns.Cluster == "") {
```

Comment:
DefaultClusterID is hardcoded to "Kubernetes", but in fact "install-istio-via-istioctl.sh" can take any cluster name, and in QE env it is set "cluster-default", so the tests are still failing.
Suggestion to change 'DefaultClusterID' here to 'config.Get().KubernetesConfig.ClusterName'

Reply:
making changes as experimental in https://github.com/kiali/kiali/pull/6101 and checking on OCP cluster, now more failures, investigating

Reply:
@jshaughn if you make the changes then we can merge this PR, it blocks other PRs.

Reply:
Yes, the tests were passing on my system because DefaultClusterID was correct, because I was using the default.  But the main issue here is bigger, we are going to get into trouble with these three different things going on.

Reply:
I'm going to look at this a bit more.  I still think the solution is to get the proper home cluster name into the cache.  It should not be "".